.github/workflows/spelling.yaml

```diff
@@ -60,8 +87,35 @@ jobs:
 
       - name: Checkout Code 🛎
         uses: actions/checkout@v4
+        if: github.event_name != 'push'
         with:
           token: ${{ steps.github-token.outputs.token }}
+          fetch-depth: 1
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Checkout Code 🛎
+        uses: actions/checkout@v4
+        if: github.event_name == 'push'
+        with:
+          token: ${{ steps.github-token.outputs.token }}
+
+      - name: Get commit message 💬
+        id: get-commit-message
+        run: |
+          git config --global --add safe.directory $(pwd)
+          export head_commit_message="$(git show -s --format=%B | tr '\r\n' ' ' | tr '\n' ' ')"
+          echo "head_commit_message=$(git show -s --format=%B | tr '\r\n' ' ' | tr '\n' ' ')" >> $GITHUB_OUTPUT
+          echo "head_commit_message = $(git show -s --format=%B | tr '\r\n' ' ' | tr '\n' ' ')"
+          if [[ $head_commit_message == *"[skip spelling]"* ]]; then
+            echo "Skip instruction detected - cancelling the workflow."
+            curl -s -LJ -o gh.tar.gz https://github.com/cli/cli/releases/download/v2.44.1/gh_2.44.1_linux_amd64.tar.gz
+            tar -xzf gh.tar.gz --strip-components 2
+            ./gh run cancel ${{ github.run_id }}
+            ./gh run watch ${{ github.run_id }}
+          fi
```

Comment:
@cicdguy I found another method of cancelling workflow for a commit which has skip instructions. It is based on [this answer](https://stackoverflow.com/a/75809743).

Here is an example [workflow run](https://github.com/walkowif/teal/actions/runs/8139073079/job/22241386681). The downside is that the usual CI image has to be pulled. The advantage is that in the PR check list, the run appears as cancelled.

![image](https://github.com/insightsengineering/r.pkg.template/assets/59475134/0925f537-c2c0-489c-b311-4e5949ce527c)

Reply:
Okay, as discussed, I left the logic to download `gh` CLI only in case the skip instructions are detected. You can take a look if overall the changes look good, and if so I'll add them to other workflows which should support skip instructions (but probably on Wednesday).